### DISCUSSION TITLE
Automatically fill in method docstrings (and even new methods/parameters?) from the Twitter API Docs

### DISCUSSION THREAD

**Discussion by Aaron1011** [tweepy/tweepy#1594] [Ideas]
I was inspired by @jackdied's PR: https://github.com/tweepy/tweepy/pull/259, which scrapes the docs, looking for mismatches with allowed_parameters (I don't think it works with the current docs/code).

This would allow Tweepy to be updated quickly whenever Twitter changes their API. It would also save us the work of documenting every single API method manually.

The downside is that dowloading all of the docs and scraping them can be slow, and the documentation HTML isn't versioned (obviously :smile:), so the script could break suddenly.

@joshthecoder: What are you thoughts?

**Comment by joshthecoder** [comment]
The tool might be useful for the current design of Tweepy's binding system, but yes it could be brittle. I have been thinking about just removing allowed_parameters all together. The only issue is for positional arguments. Ex: api.update_status(text) needs to map the 1st argument to 'text'.
